pkg/consensus/recovery_message.go

```diff
@@ -68,7 +71,16 @@ func (m *recoveryMessage) DecodeBinary(r *io.BinReader) {
 	}
 
 	r.ReadArray(&m.preparationPayloads)
-	r.ReadArray(&m.commitPayloads)
+	lu := r.ReadVarUint()
+	if lu > io.MaxArraySize {
```

Comment:
Doesn't it have some different limit in C# node (I've not checked yet, but these messages usually are more limited)?

Reply:
`Blockchain.MaxValidators`

Reply:
Fixed.